external/upstream/libxc/CMakeLists.txt

```diff
@@ -25,6 +25,7 @@ else()
                    -DCMAKE_INSTALL_LIBDIR=${CMAKE_INSTALL_LIBDIR}
                    -DCMAKE_INSTALL_BINDIR=${CMAKE_INSTALL_BINDIR}
                    -DCMAKE_INSTALL_INCLUDEDIR=${CMAKE_INSTALL_INCLUDEDIR}
+                   -DCMAKE_OSX_SYSROOT=${CMAKE_OSX_SYSROOT}
```

Comment:
Why the libxc cmake change? (asking because it is unrelated to the PR topic)

Reply:
Because previously `OSX_SYSROOT` did not get picked up be the Psi4 ExternalProject & friends, so instead of using the one I defined via `cmake`, it picked up the one from the system. Don't know if there could be any further problems from this change @loriab, @susilehtola?

Reply:
There shouldn't be any problems sending `OSX_SYSROOT` through. The need might not have gotten picked up on before b/c internal builds are less common on Macs.

Reply:
FWIW, my experience with Mac builds has been that they haven't worked out-of-the-box for the last couple months. While initial signs pointed to exactly the SDK problem that Max's fix suggests, I've only been able to tame it by passing in system clang rather than conda clang.